packages/components/src/color-palette/index.js

```diff
@@ -125,6 +127,45 @@ export function CustomColorPickerDropdown( { isRenderedInSidebar, ...props } ) {
 	);
 }
 
+const extractColorNameFromCurrentValue = (
+	currentValue,
+	colors = [],
+	showMultiplePalettes = false
+) => {
+	if ( ! currentValue ) {
+		return '';
+	}
+
+	let colorNameToReturn;
+
+	if ( showMultiplePalettes ) {
+		for ( const { colors: colorPaletteColors } of colors ) {
```

Comment:
Non-blocking nit: Because the body of these loops are the same in both conditions, I wonder if we should normalize the arrays earlier to simplify the code? For example:

```js
const colorPalettes = showMultiplePalettes ? colors : [ { colors } ];

for ( const { colors: colorPaletteColors } of colorPalettes ) {
	for ( const {
		name: colorName,
		color: colorValue,
	} of colorPaletteColors ) {
		if (
			colord( currentValue ).toHex() ===
			colord( colorValue ).toHex()
		) {
			return colorName;
		}
	}
}
```